covalent_ui/webapp/src/components/terminal/Terminal.js

```diff
@@ -0,0 +1,67 @@
+
+
+import React, { useRef, useEffect, memo } from 'react'
+import { XTerm } from 'xterm-for-react'
+import { FitAddon } from 'xterm-addon-fit';
+import io from 'socket.io-client'
+import './index.css';
+const Terminal = () => {
+    const xtermRef = useRef()
+    const fitAddon = new FitAddon();
+    const socket = io(process.env.REACT_APP_SOCKET_URL, {
```

Comment:
This would get called on re-renders, however I believe the docs say that the connection manager is re-used but I would double check this won't have any side effects https://socket.io/docs/v4/client-api/#iourl otherwise placing in useEffect -> set manager with useState might make sense

Reply:
Double checked and it seemed to use the same socket connection on re-renders as well.